### DISCUSSION TITLE
一个关于复选框的现象

### DISCUSSION THREAD

**Discussion by HuangLei0x2e5** [layui/layui#2445] [讨论]
我有一个复选框

            <input type="checkbox" id='a' title="供应商A">
            
           我在页面加载时，执行这样一行
           layui.$('#a').prop('checked', true);
           
           input会被选上
           
           但是如果我再写一个button ,在button的onclick事件函数中执行这样一句的话，就没有效果
           
           不知道为什么

**Comment by LSL1618** [comment]
页面加载时更改元素属性值，会跟着表单元素一起会被美化渲染。如果用按钮的点击事件去更改元素属性值，由于没有被重新渲染会导致没有选中效果。
